src/item.cpp

```diff
@@ -5332,12 +5332,18 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
                              qty, m.second.flags() ) );
             }
 
-            // non-auxiliary gunmods may provide additional modes for the base item
+            // non-auxiliary gunmods may provide additional modes for the base item
+
+
         } else if( e->is_gunmod() ) {
-            for( auto m : e->type->gunmod->mode_modifier ) {
-                res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
+            for( auto m : e->type->gunmod->mode_modifier )
+                  if( m.first == ("REACH"))   {
```

Comment:
The () around "REACH" are unnecessary, and each '(' or ')' should be separated from other characters by a space on each side.  Also, there should be only one space between the final ')' and the '{'.